bcache/get.go

```diff
@@ -0,0 +1,310 @@
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		err = parsePriorityStats(scanner.Text(), &res)
+		if err != nil {
+			p.err = fmt.Errorf("failed to parse: %s (%s)", path, err)
+			return res
+		}
+	}
+	if err := scanner.Err(); err != nil {
+		p.err = fmt.Errorf("failed to parse: %s (%s)", path, err)
+		return res
+	}
+	return res
+}
+
+// GetStats collects from sysfs files data tied to one bcache ID.
+func GetStats(uuidPath string) (*Stats, error) {
+	var (
+		// stats = map[string]float64{}
+		bs Stats
+	)
+
+	par := parser{ uuidPath: uuidPath}
+
+	// bcache stats
+
+	// dir <uuidPath>
+	par.setSubDir("")
+	bs.Bcache.AverageKeySize = par.readValue("average_key_size")
+	bs.Bcache.BtreeCacheSize = par.readValue("btree_cache_size")
+	bs.Bcache.CacheAvailablePercent = par.readValue("cache_available_percent")
+	bs.Bcache.Congested = par.readValue("congested")
+	bs.Bcache.RootUsagePercent = par.readValue("root_usage_percent")
+	bs.Bcache.TreeDepth = par.readValue("tree_depth")
+
+	// bcache stats (internal)
+
+	// dir <uuidPath>/internal
+	par.setSubDir("internal")
+	bs.Bcache.Internal.ActiveJournalEntries = par.readValue("active_journal_entries")
+	bs.Bcache.Internal.BtreeNodes = par.readValue("btree_nodes")
+	bs.Bcache.Internal.BtreeReadAverageDurationNanoSeconds = par.readValue("btree_read_average_duration_us")
+	bs.Bcache.Internal.CacheReadRaces = par.readValue("cache_read_races")
+
+	// bcache stats (period)
+
+	// dir <uuidPath>/stats_five_minute
+	par.setSubDir("stats_five_minute")
+	bs.Bcache.FiveMin.Bypassed = par.readValue("bypassed")
+	bs.Bcache.FiveMin.CacheHits = par.readValue("cache_hits")
+
+	bs.Bcache.FiveMin.Bypassed = par.readValue("bypassed")
+	bs.Bcache.FiveMin.CacheBypassHits = par.readValue("cache_bypass_hits")
+	bs.Bcache.FiveMin.CacheBypassMisses = par.readValue("cache_bypass_misses")
+	bs.Bcache.FiveMin.CacheHits = par.readValue("cache_hits")
+	bs.Bcache.FiveMin.CacheMissCollisions = par.readValue("cache_miss_collisions")
+	bs.Bcache.FiveMin.CacheMisses = par.readValue("cache_misses")
+	bs.Bcache.FiveMin.CacheReadaheads = par.readValue("cache_readaheads")
+
+	// dir <uuidPath>/stats_total
+	par.setSubDir("stats_total")
+	bs.Bcache.Total.Bypassed = par.readValue("bypassed")
+	bs.Bcache.Total.CacheHits = par.readValue("cache_hits")
+
+	bs.Bcache.Total.Bypassed = par.readValue("bypassed")
+	bs.Bcache.Total.CacheBypassHits = par.readValue("cache_bypass_hits")
+	bs.Bcache.Total.CacheBypassMisses = par.readValue("cache_bypass_misses")
+	bs.Bcache.Total.CacheHits = par.readValue("cache_hits")
+	bs.Bcache.Total.CacheMissCollisions = par.readValue("cache_miss_collisions")
+	bs.Bcache.Total.CacheMisses = par.readValue("cache_misses")
+	bs.Bcache.Total.CacheReadaheads = par.readValue("cache_readaheads")
+
+	if par.err != nil {
+		return nil, par.err
+	}
+
+	// bdev stats
+
+	reg := path.Join(uuidPath, "bdev[0-9]*")
+	bdevDirs, err := filepath.Glob(reg)
+	if err != nil {
+		return nil, err
+	}
+	bs.Bdevs = []bdevStats{}
```

Comment:
Nit: since you already know the necessary capacity for this slice (`len(bdevDirs)`), initialize it with that capacity immediately.